packages/netlify-cms-backend-github/src/API.js

```diff
@@ -557,7 +557,7 @@ export default class API {
     const contentKey = slug;
     const branchName = this.generateBranchName(contentKey);
     return this.retrieveMetadata(contentKey)
-    .then(metadata => this.closePR(metadata.pr, metadata.objects))
```

Comment:
Looks like we _should_ be removing associated metadata when a PR is closed, but we're not.

Reply:
@erquhart Feel free to push a fix for this here, or separately.